python/lsst/dax/imgserv/getimage/imageget.py

```diff
@@ -0,0 +1,625 @@
+        if bbox:
+            image = self._butler.get(self._imagedataset_sub(), bbox=bbox,
+                                     dataId=data_id, immediate=True)
+        else:
+            image = self._butler.get(self._imagedataset_type, dataId=data_id)
+        return image
+
+    def _imagedataset_md(self):
+        # Return the butler policy name to retrieve metadata
+        return self._imagedataset_type + "_md"
+
+    def _imagedataset_sub(self):
+        # Return the dataset type for sub-images
+        return self._imagedataset_type + "_sub"
+
+
+    def _cutout_from_src(self, src_image, xy_center_x, xy_center_y, width,
+                         height, wcs):
+        # Returns an image cutout from the source image.
+        # srcImage - Source image.
+        # xy_center - The center of region to cutout in pixels.
+        # width - The width in pixels.
+        # height - The height in pixels.
+        # height and width trimmed if they go past the edge of source image.
+        # First, center the cutout image.
+        pix_ulx = int(xy_center_x - width / 2.0)
+        pix_uly = int(xy_center_y - height / 2.0)
+        xy_center = afwGeom.Point2I(pix_ulx, pix_uly)
+        log.debug("xy_center={}".format(xy_center))
+        src_box = src_image.getBBox()
+        # assuming both src_box and xy_center to be in Box2I
+        co_box = afwGeom.Box2I(xy_center,
+                               afwGeom.Extent2I(int(width), int(height)))
+        if co_box.overlaps(src_box):
+            co_box.clip(src_box)
+        else:
+            self._log.debug(
+                "cutout image wanted is OUTSIDE source image -> None")
+            return None
+        if isinstance(src_image, afwImage.ExposureF):
+            self._log.debug(
+                "co_box pix_ulx={} pix_end_x={} pix_uly={} pix_end_y={}"
+                .format(pix_ulx, pix_ulx + width, pix_uly, pix_uly + height))
+            # image will keep wcs from source image
+            cutout = afwImage.ExposureF(src_image, co_box)
+        elif isinstance(src_image, afwImage.ExposureU):
+            cutout = afwImage.ExposureU(src_image, co_box)
+        else:
+            raise Exception("Unexpected source image object type")
+        return cutout
+
+    def _data_id_from_qr(self, q_results):
+        # identify and fetch the data ID from 1 of 2 defined sets
+        ln = q_results[0]
+        # first try run, camcol, field, filter keys
+        if self._butler_keys == sorted(["run", "camcol", "field", "filter"]):
+            run, camcol, field, filtername = ln[2:6]
+            data_id = {"run": run, "camcol": camcol, "field": field,
+                       "filter": filtername}
+            return data_id
+        # if no match, then try tract, patch, filter keys
+        if self._butler_keys == sorted(["tract", "patch", "filter"]):
+            tract, patch, filtername = ln[2:5]
+            data_id = {"tract": tract, "patch": patch, "filter": filtername}
```

Comment:
This looks strikingly similar to what's on line :306 in this file with the keys... can this be used instead?